src/sensor_lis2dw.c

```diff
@@ -6,32 +6,53 @@
 // This file may be distributed under the terms of the GNU GPLv3 license.
 
 #include <string.h> // memcpy
+#include "autoconf.h" // CONFIG_HAVE_GPIO_SPI
+#include "board/gpio.h" // irq_disable
 #include "board/irq.h" // irq_disable
 #include "board/misc.h" // timer_read_time
 #include "basecmd.h" // oid_alloc
 #include "command.h" // DECL_COMMAND
 #include "sched.h" // DECL_TASK
 #include "sensor_bulk.h" // sensor_bulk_report
 #include "spicmds.h" // spidev_transfer
+#include "i2ccmds.h" // i2cdev_s
 
 #define LIS_AR_DATAX0 0x28
 #define LIS_AM_READ   0x80
+#define LIS_MS_SPI    0x40
+#define LIS_MS_I2C    0x80
 #define LIS_FIFO_SAMPLES 0x2F
 
 #define BYTES_PER_SAMPLE 6
 
 struct lis2dw {
     struct timer timer;
     uint32_t rest_ticks;
-    struct spidev_s *spi;
+    void *bus;
```

Comment:
As a minor note, this might be a little more readable as `union { struct spidev_s *spi ; struct i2c_config *i2c; };`  Not a big deal though.

Reply:
Honestly, I just thought about `void *` when I was implementing this, and sort of forgot that `union` existed! I can change it when this gets refactored after #6689 gets merged. Looks like the I2C reads will change.

Reply:
Changed in the new revision.